### PR TITLE
[flutter_tool] support --use-application-binary in flutter drive

### LINKED ISSUE #56604: Un-hide --use-application-binary option for flutter run/drive
Using this, users that don't use Gradle or Xcode can by pass the automatic building for APK/IPAs, or use custom built APKs with options that `flutter run` doesn't support.

We should expose and document this option, including cases where it might not work.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

--use-application-binary allows running with an already built APK. This can be useful for speeding up CI test cases, or in our case eventually supporting some sort of build server.

Demonstrate that this works by updating the old gallery test to use it

Fixes https://github.com/flutter/flutter/issues/56604

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 1]
```diff
@@ -466,8 +466,14 @@ Future<LaunchResult> _startApp(
   globals.printTrace('Stopping previously running application, if any.');
   await appStopper(command);
 
-  final ApplicationPackage package = await command.applicationPackages
-    .getPackageForPlatform(await command.device.targetPlatform, buildInfo: command.getBuildInfo());
+  final File applicationBinary = command.stringArg('use-application-binary') == null
+    ? null
+    : globals.fs.file( command.stringArg('use-application-binary'));
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/drive.dart:471]:
```suggestion
    : globals.fs.file(command.stringArg('use-application-binary'));
```

**Review by jmagman — APPROVED** [review]
Why was this hidden in the first place?

**Comment by jonahwilliams** [comment]
Originally it was only for google3, and it wasn't ever added to drive since drive predates it by quite a few years.

**Review by christopherfujino — APPROVED** [review]
lgtm

**Pull request merged**